### PR TITLE
beta3 release changes

### PR DESCRIPTION
**Pull request by mhowlett**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,21 +3,23 @@
...
+  - All `event`s on the client classes have been replaced with corresponding `Set...Handler` methods on the builder classes.
+    - This allows (enforces) handlers are set on librdkafka initialization (which is important for some handlers, particularly the log handler).
+    - `event`s allow for more than one handler to be set, but this is often not appropriate (e.g. `OnPartitionsAssigned`), and never necessary. This is no longer possible.
+    - `event`s are also not async friendly (handlers can't return `Task`). The Set...Handler appropach can be extend in such a way that it is.
 - Avro serdes no longer make blocking calls to `ICachedSchemaRegistryClient` - everything is `await`ed.
-- References librdkafka.redist [1.0.0-RC5](https://github.com/edenhill/librdkafka/releases/tag/v1.0.0-RC5)
+  - Note: The `Consumer` implementation still calls async deserializers synchronously because the `Consumer` API is still otherwise fully synchronous.
+- Reference librdkafka.redist [1.0.0-RC7](https://github.com/edenhill/librdkafka/releases/tag/v1.0.0-RC7)
+  - Notable features: idempotent producer, sparse connections.
```
Comment (edenhill) [inline CHANGELOG.md]: KIP-62 max.poll.interval.ms

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,21 +3,23 @@
...
   - Note: End of partition notification is now disabled by default (enable using the `EnablePartitionEof` config property).
-- Removed `Consumer.OnPartitionEOF` in favor of `ConsumeResult.IsPartitionEOF`.
+- Removed the `Consumer.OnPartitionEOF` event in favor notifying of partition eof via `ConsumeResult.IsPartitionEOF`.
```
Comment (edenhill) [inline CHANGELOG.md]: They are not emitted by default anymore by librdkafka, you should instruct the user to enable them if they want that event.

File: README.md [hunk 3]
```diff
@@ -219,7 +225,7 @@ For more information about working with Avro in .NET, refer to the the blog post
 
 ### Error Handling
 
-Errors raised via a client's `OnError` event should be considered informational except when the `IsFatal` flag
+Errors delivered to a client's error handler should be considered informational except when the `IsFatal` flag
```
Comment (edenhill) [inline README.md]: `ErrorHandler` to give a hint that it relates to SetErrorHandler

File: examples/Producer/Producer.csproj [hunk 4]
```diff
@@ -9,7 +9,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <!-- nuget package reference: <PackageReference Include="Confluent.Kafka" Version="1.0-beta2" /> -->
+    <!-- nuget package reference: <PackageReference Include="Confluent.Kafka" Version="1.0.0-beta3" /> -->
```
Comment (edenhill) [inline examples/Producer/Producer.csproj]: perfect!

File: CHANGELOG.md (hunk 2 above)
Comment (mhowlett) [reply CHANGELOG.md]: yep, that's also noted in the changelog.

**Pull request merged**